scenarios/csharp/dotnet/captioning/Program.cs

```diff
@@ -0,0 +1,423 @@
+            {
+                languageIDLanguages = languageIDLanguagesResult.Split(',');
+            }
+            
+            return new UserConfig(
+                CmdOptionExists(args, "-f"),
+                CmdOptionExists(args, "-m"),
+                languageIDLanguages,
+                GetCmdOption(args, "-i"),
+                GetCmdOption(args, "-o"),
+                GetCmdOption(args, "-p"),
+                CmdOptionExists(args, "-q"),
+                CmdOptionExists(args, "-u"),
+                GetCmdOption(args, "-r"),
+                CmdOptionExists(args, "-s"),
+                CmdOptionExists(args, "-t"),
+                args[args.Length - 2],
+                args[args.Length - 1]
+            );
+        }
+
+        static AudioConfig AudioConfigFromUserConfig(UserConfig userConfig)
+        {
+            if (userConfig.inputFile is string inputFileValue)
+            {
+                return AudioConfig.FromWavFileInput(inputFileValue);
+            }
+            else
+            {
+                return AudioConfig.FromDefaultMicrophoneInput();
+            }
+        }
+
+        static SpeechConfig SpeechConfigFromUserConfig(UserConfig userConfig)
+        {
+            SpeechConfig speechConfig;
+            if (userConfig.languageIDLanguages is not null)
+            {
+                speechConfig = SpeechConfig.FromEndpoint(new Uri(V2EndpointFromRegion(userConfig.region)), userConfig.subscriptionKey);
+            }
+            else
+            {
+                speechConfig = SpeechConfig.FromSubscription(userConfig.subscriptionKey, userConfig.region);
+            }
+
+            if (userConfig.profanityFilterRemoveEnabled)
+            {
+                speechConfig.SetProfanity(ProfanityOption.Removed);
+            }
+            else if (userConfig.profanityFilterMaskEnabled)
+            {
+                speechConfig.SetProfanity(ProfanityOption.Masked);
+            }
+            
+            if (userConfig.stablePartialResultThreshold is string stablePartialResultThresholdValue)
+            {
+                speechConfig.SetProperty(PropertyId.SpeechServiceResponse_StablePartialResultThreshold, stablePartialResultThresholdValue);
+            }
+            
+            if (userConfig.trueTextEnabled)
+            {
+                speechConfig.SetProperty(PropertyId.SpeechServiceResponse_PostProcessingOption, "TrueText");
+            }
+            
+            return speechConfig;
+        }
+
+        static SpeechRecognizer SpeechRecognizerFromSpeechConfig(SpeechConfig speechConfig, AudioConfig audioConfig, UserConfig userConfig)
+        {
+            SpeechRecognizer speechRecognizer;
+            if (userConfig.languageIDLanguages is string[] languageIDLanguagesValue)
+            {
+                AutoDetectSourceLanguageConfig detectLanguageConfig = AutoDetectSourceLanguageConfig.FromLanguages(languageIDLanguagesValue);
```

Comment:
Matches the naming convention already in our GitHub samples.
```suggestion
                AutoDetectSourceLanguageConfig autoDetectSourceLanguageConfig = AutoDetectSourceLanguageConfig.FromLanguages(languageIDLanguagesValue);
```